### PR TITLE
Use global variable for directoryKey in global resources

### LINKED ISSUE #4683: Cloning Preferences results in a 404 error
**Describe the bug**
Cloning any Preference (Global/Remote) in Global Resources causes a 404 error displayed to the user.

**To Reproduce**
1. Go to App Resources -> Global Resources -> Any Preference (e.g. Global Preferences; https://fwri1924-edge.test.specifysystems.org/specify/resources/app-resource/1/)
2. Click on Clone
4. See 404 error

Note: found bug while testing #4531

**Expected behavior**
Page should display the selected Resource cloned with a unique name.
Behavior in `v7.9`:

https://github.com/specify/specify7/assets/118556884/997ae1f0-c9ab-4c83-9287-7aaf8d7659fe

**Evidence**

https://github.com/specify/specify7/assets/118556884/eb136268-585e-4a2f-b2fb-1851cb4fd51b

OS: macOS Monterey (12.6.5)
Browser: Chrome
Specify 7 Version: `edge`
Database Name: `fwri_1_9_24`

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4683

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

- Go to App Resources -> Global Resources -> Any Preference (e.g. Global Preferences; https://fwri1924-edge.test.specifysystems.org/specify/resources/app-resource/1/)
- Click on Clone
- [ ] Verify the resource gets cloned. 
- [ ] proceed with several different type of resources

### CODE REVIEW

**Comment by CarolineDenis** [comment]
NOTES / TODO: 
- this is a first version to see if it works and propose something rather than nothing. 
- need to clean the code

**Review by emenslin — APPROVED** [review]
### Testing instructions

- Go to App Resources -> Global Resources -> Any Preference (e.g. Global Preferences; https://fwri1924-edge.test.specifysystems.org/specify/resources/app-resource/1/)
- Click on Clone
- [x] Verify the resource gets cloned. 
- [x] proceed with several different type of resources 

Looks good! 

https://github.com/specify/specify7/assets/150940840/1e9a4cec-cb96-4d93-8722-f9f18e490616

File: specifyweb/frontend/js_src/lib/components/AppResources/EditorWrapper.tsx [hunk 1]
```diff
@@ -92,16 +92,20 @@ export function Wrapper({
...
-              directory.id
-            ),
+            directoryKey: Array.isArray(directory)
+              ? findAppResourceDirectoryKey(
+                  resourcesTree,
+                  directory.find(
+                    (dir) => dir.resource_uri === record.spAppResourceDir
+                  )?.id || -1
+                )
+              : findAppResourceDirectoryKey(resourcesTree, directory.id),
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/AppResources/EditorWrapper.tsx:108]: Good job finding the issue.
But I would propose a much simpler solution

```suggestion
            directoryKey:
              directory.scope === 'global'
                ? globalResourcesKey
                : resourcesTreefindAppResourceDirectoryKey(
                    resourcesTree,
                    directory.id
                  ),

```

Where `globalResourcesKey` should be `export const globalResourcesKey = 'globalResources';`, and should be defined in tree.ts, and then also used in this place:

https://github.com/specify/specify7/blob/1858540f8c16eac398f39cca870abb55e346baa8/specifyweb/frontend/js_src/lib/components/AppResources/tree.ts#L40

by having a variable for this, any changes to the key would automatically update the places where it's used

and speaking of changes, feel free to rename the key value from `globalResources` to `globalResource` to better describe it's purpose (as there is only "one" global app resource)

Then, undo all other changes in this PR

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/AppResources/EditorWrapper.tsx (hunk 1 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/AppResources/EditorWrapper.tsx]: so much easier!!!

**Review by maxpatiiuk — APPROVED** [review]

**Pull request merged**